tests/publiccloud/check_registercloudguest.pm

```diff
@@ -44,7 +44,23 @@ sub run {
         return;
     }
 
-    if (is_container_host()) {
+    # this is special use case needed by dev ops team within SUSE. Should not be used for any product testing.
+    # Use this variable ONLY if you really know what you are doing!
+    if (check_var('PUBLIC_CLOUD_INFRA', '1')){
+        my $prefix = "PREPARATION PHASE.";
+        my $rmt_ipv4 = get_required_var("PUBLIC_CLOUD_INFRA_RMT_V4");
+        my $rmt_ipv6 = get_required_var("PUBLIC_CLOUD_INFRA_RMT_V6");
+        if(is_ondemand()) {
+            check_instance_unregistered($instance, "$prefix BYOS instance was registered before we started");
+            $instance->ssh_assert_script_run(cmd => "sudo ${path}registercloudguest --clean", fail_message => "$prefix registercloudguest --clean failed");
```

Comment:
```suggestion
            $instance->ssh_assert_script_run("sudo ${path}registercloudguest --clean", fail_message => "$prefix registercloudguest --clean failed");
```

Reply:
hm , I am not sure . For me it looks more verbose with `cmd => `

Reply:
and so you do assert_script_run(cmd => 'ls /'); as well? No, you just do assert_script_run('ls /');.

Reply:
this discussion is really strange to me . Function support both calls with `cmd` and without `cmd` . I prefer to do it with `cmd` . Also I prefer when each and every function call has `()` but because perl allowing to skip I am ignoring the fact that people randomly deciding to put it or not . We don't have code policy which not allow me to use `cmd` so do you have some certain reason why I can not use `cmd` except that you don't like it ?